app/main.py

```diff
@@ -1 +1,47 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.name = name
```

Comment:
Redundant assignment: The `self.name = name` line is unnecessary here because the `super().__init__(name)` call in line 12 already initializes `self.name` in the `SoftwareEngineer` base class.